### PR TITLE
update CAPI discussion link

### LINKED ISSUE #505: Link-check / lint_docs test failing because of a broken link
The README.md file contains a broken link [https://discuss.kubernetes.io/c/contributors/cluster-api](https://discuss.kubernetes.io/c/contributors/cluster-api) at `line 129` which is referencing to Kubernetes discuss page at `line 81`.

I don't see this particular link in other providers so it's better to remove it(both these lines) as because of it, the `Link-check/link_docs` test is failing.

### PR DESCRIPTION
**Pull request by Aayush987**
/kind documentation

**What this PR does / why we need it**:
It fixes the issue with the broken link in README.md 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes  #505

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```NONE

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,7 +78,6 @@ If you have an active interest and you want to get involved, you have real power
 ### Office hours
 
 - Join the [SIG Cluster Lifecycle](https://groups.google.com/forum/#!forum/kubernetes-sig-cluster-lifecycle) Google Group for access to documents and calendars.
-- Participate in the conversations on [Kubernetes Discuss][kubernetes discuss]
```
Comment (cpanato) [inline README.md:81]: i would not remove the link but instead update to the correct/updated location: https://groups.google.com/forum/#!forum/kubernetes-sig-cluster-lifecycle

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks for updating the documentation

File: README.md (hunk 1 above)
Comment (jayesh-srivastava) [reply README.md:81]: @Aayush987 please make the required modifications.

Comment (Aayush987) [reply README.md:81]: Ok Thank you for the suggestion

Comment (jayesh-srivastava) [reply README.md]: @Aayush987 looks good but can you please squash these commits into one. Three commits are uneccesary.

Comment (Aayush987) [reply README.md]: @jayesh-srivastava  I actually dont know how to squash commits. Can you tell me how to do it?

Comment (cpanato) [reply README.md]: that is fine for now @Aayush987 i can squash the commits in the merge but for future PRs you can read https://github.com/kubernetes/community/blob/f8158c06a4293b48f3364fd2af76d5f4a2915f0f/contributors/guide/contributor-cheatsheet/README.md#squashing-commits

**Review by cpanato — APPROVED** [review]
thanks!
/approve
/lgtm

**Pull request merged**